inc/themes/core.base/frontend/templates/index/index.twig

```diff
@@ -32,6 +32,14 @@
                         {% if user.invisible == 1 and (mybb.usergroup.canviewwolinvis == 1 or user.uid == mybb.user.uid) %}*{% endif %}
                     {% endfor %}
                 </p>
+                {% if mybb.settings.showgroupslegend %}
```

Comment:
```suggestion
                {% if mybb.settings.showgroupslegend and groups %}
```

Otherwise, some HTML will be added even if there are no groups to display.

Why do we need a global setting for this again ? I think the global setting to be unnecessary now.

Reply:
If you want to disable the legend entirely and have many groups, it's probably easier to just toggle the global setting.